packages/azure-http-specs/specs/azure/resource-manager/resources/mockapi.ts

```diff
@@ -151,17 +163,99 @@ const validResourceExtensionsResource = {
   },
 };
 
+function getServiceKey(resourceUri: string) {
+  switch (resourceUri) {
+    case RESOURCE_GROUP_SCOPE_URI:
+      return "ResourceGroup" as const;
+    case SUBSCRIPTION_SCOPE_URI:
+      return "Subscription" as const;
+    case TENANT_SCOPE_URI:
+      return "Tenant" as const;
+    case RESOURCE_SCOPE_URI:
+      return "Resource" as const;
+    default:
+      throw new ValidationError(
+        "Invalid resource uri",
+        `${RESOURCE_GROUP_SCOPE_URI} | ${SUBSCRIPTION_SCOPE_URI} | ${TENANT_SCOPE_URI} | ${RESOURCE_SCOPE_URI}`,
+        resourceUri,
+      );
+  }
+}
+
+function requestHandler(
+  req: MockRequest,
+  requestMethod: string,
+  resourceUri: string,
+  validResource: any,
+) {
+  const serviceKey = getServiceKey(resourceUri);
+  switch (requestMethod) {
+    case "get":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json(validResource),
+      } as const;
+    case "put":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json(validResource),
+      } as const;
+    case "patch":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json({
+          ...validResource,
+          properties: {
+            provisioningState: "Succeeded",
+            description: "valid2",
+          },
+        }),
+      } as const;
+    case "delete":
+      return {
+        pass: serviceKey,
+        status: 204,
+      } as const;
+    case "list":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json({
+          value: [validResource],
+        }),
+      } as const;
+    default:
+      throw new ValidationError(
+        "Invalid request method",
+        `"get" | "put" | "patch" | "delete" | "list"`,
+        requestMethod,
+      );
+  }
+}
+
+function getUri(resourceUri: string, extensionName: string, requestMethod: string): string {
+  switch (requestMethod) {
+    case "list":
+      return `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources`;
+    default:
+      return `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/${extensionName}`;
+  }
+}
+
 // extension tracked resource
-Scenarios.Azure_ResourceManager_Resources_ExtensionsResources_get = passOnSuccess([
+Scenarios.Azure_ResourceManager_Resources_ExtensionsResources_get = withServiceKeys([
+  "ResourceGroup",
+  "Subscription",
+  "Tenant",
+  "Resource",
+]).pass([
   {
-    uri: "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Resources/extensionsResources/:extensionName",
+    uri: getUri(RESOURCE_GROUP_SCOPE_URI, EXTENSION_RESOURCE_NAME, "get"),
```

Comment:
Why does it look like it's still checking only "//"? 
```typescript
const RESOURCE_GROUP_SCOPE_URI = `/subscriptions/${SUBSCRIPTION_ID_EXPECTED}/resourceGroups/${RESOURCE_GROUP_EXPECTED}`;

function getUri(resourceUri: string, extensionName: string, requestMethod: string): string {
  switch (requestMethod) {
    case "list":
      return `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources`;
    default:
      return `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/${extensionName}`;
  }
}
```

Did you have generated java code? Did the test pass for both cases?

Reply:
This function can generate URIs containing "//" or "/", so both conditions can be check. which has been verified by the test case.